device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -952,8 +971,8 @@ private AmqpsSessionHandler addSessionHandler(DeviceClientConfig deviceClientCon
         {
             if (existingAmqpsSessionHandler.getDeviceId().equals(deviceClientConfig.getDeviceId()))
             {
-                amqpsSessionHandler = existingAmqpsSessionHandler;
-                break;
```

Comment:
This session handler was retrieved from the local state, but later added to the same list it was retrieved from for some reason. Now we just return it once it is found. No need to add a duplicate session handler to the list of session handlers